src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingExerciseResource.java

```diff
@@ -593,4 +598,26 @@ public ProgrammingExerciseResource(ProgrammingExerciseRepository programmingExer
                     .body(null);
         }
     }
+
+    /**
+     * GET /programming-exercises/:exerciseId/is-released-and-has-results : Check if the given exercise is released and has at least one student result.
+     *
+     * @param exerciseId the id of a ProgrammingExercise
+     * @return the ResponseEntity with status 200 (OK) and true if the exercise is released and there is at least one result, false if not. Returns 404 (notFound) if the exercise does not exist.
+     */
+    @GetMapping(value = "/programming-exercises/{exerciseId}/is-released-and-has-results")
+    @PreAuthorize("hasAnyRole('TA', 'INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<Boolean> isExerciseReleasedAndHasResult(@PathVariable Long exerciseId) {
+        Optional<ProgrammingExercise> programmingExercise = programmingExerciseRepository.findById(exerciseId);
+        if (programmingExercise.isEmpty()) {
+            return notFound();
+        }
+        ZonedDateTime releaseDate = programmingExercise.get().getReleaseDate();
```

Comment:
Even if this is not an endpoint that seems to be critical from a security point, I would ask you to double check that only TAs and instructors of the course can access this with
`authCheckService.isAtLeastTeachingAssistantInCourse(course, user)`

Reply:
You are right, I added a permission check here.